src/battle_util.c

```diff
@@ -11185,3 +11194,36 @@ bool32 IsGen6ExpShareEnabled(void)
     return FlagGet(I_EXP_SHARE_FLAG);
 #endif
 }
+
+bool8 CanTargetPartner(u8 battlerDef)
+{
+    return (gBattleTypeFlags & BATTLE_TYPE_DOUBLE)
+        && IsBattlerAlive(BATTLE_PARTNER(battlerDef))
+        && battlerDef != BATTLE_PARTNER(gBattlerAttacker);
+}
+
+bool8 TargetFullyImmuneToCurrMove(u8 battlerDef)
+{
+    u8 moveType = 0;
+    GET_MOVE_TYPE(gCurrentMove, moveType);
+
+    return (CalcTypeEffectivenessMultiplier(gCurrentMove, moveType, gBattlerAttacker, battlerDef, FALSE) & MOVE_RESULT_NO_EFFECT)
+         || IsBattlerProtected(battlerDef, gCurrentMove)
+         || IsSemiInvulnerable(battlerDef, gCurrentMove)
+         || DoesTargetHaveAbilityImmunity();
+}
+
+// For Dragon Darts
+static bool8 DoesTargetHaveAbilityImmunity(void)
+{
+    BattleScriptPushCursor(); // Backup the current script
+
+    if (AbilityBattleEffects(ABILITYEFFECT_MOVES_BLOCK, gBattlerTarget, 0, 0, 0)
+    || AbilityBattleEffects(ABILITYEFFECT_ABSORBING, gBattlerTarget, 0, 0, 0))
```

Comment:
@BuffelSaft do you know why `DoesTargetHaveAbilityImmunity` checks for current target but all other checks in `TargetFullyImmuneToCurrMove` are for current target partner? If I understand it correctly then we are checking if new potential  target would be immune.